diff --git a/hyperon-space/src/index/trie.rs b/hyperon-space/src/index/trie.rs
index 1fd98fb..e4db9b4 100644
--- a/hyperon-space/src/index/trie.rs
+++ b/hyperon-space/src/index/trie.rs
@@ -542,7 +542,7 @@ impl TrieKey {
 
     #[inline]
     fn value(&self) -> usize {
-        self.0 & TK_VALUE_MASK - TK_MAX_EXPRESSION_SIZE
+        (self.0 & TK_VALUE_MASK) - TK_MAX_EXPRESSION_SIZE
     }
 
     #[inline]
